File: PESDK_LITE(OpenSource)_Android_220623_1.5.3.520278/PEUISDK/src/main/java/com/pesdk/uisdk/widget/SquareFrameLayout.java
package com.pesdk.uisdk.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.FrameLayout;

/**
 * 区域正方形
 */
public class SquareFrameLayout extends FrameLayout {

    public SquareFrameLayout(Context context) {
        this(context, null, 0);
    }

    public SquareFrameLayout(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public SquareFrameLayout(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int previewWidth = MeasureSpec.getSize(widthMeasureSpec);

        super.onMeasure(
                MeasureSpec.makeMeasureSpec(previewWidth, MeasureSpec.EXACTLY),
                MeasureSpec.makeMeasureSpec(previewWidth, MeasureSpec.EXACTLY));
    }

}

File: PESDK_LITE(OpenSource)_Android_220623_1.5.3.520278/PEAlbum/src/main/java/com/pesdk/album/uisdk/widget/SquareFrameLayout.kt
package com.pesdk.album.uisdk.widget

import android.content.Context
import android.util.AttributeSet
import android.widget.FrameLayout

/**
 * 区域正方形
 */
class SquareFrameLayout @JvmOverloads constructor(
    context: Context, attrs: AttributeSet? = null,
    defStyle: Int = 0
) : FrameLayout(
    context, attrs, defStyle
) {
    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val previewWidth = MeasureSpec.getSize(widthMeasureSpec)
        super.onMeasure(
            MeasureSpec.makeMeasureSpec(previewWidth, MeasureSpec.EXACTLY),
            MeasureSpec.makeMeasureSpec(previewWidth, MeasureSpec.EXACTLY)
        )
    }
}